CMakeLists.txt

```diff
@@ -19,6 +19,15 @@ set(VERSION_BUILD ${BUILD_NUMBER})
 # Used for write_basic_package_version_file()
 project(triton VERSION ${VERSION_MAJOR}.${VERSION_MINOR})
 
+
+if(CMAKE_HOST_SYSTEM_NAME MATCHES "Windows")
+    if (CMAKE_CXX_COMPILER_ID STREQUAL "Clang" AND CMAKE_GENERATOR_TOOLSET STREQUAL "ClangCL")
+        set(CMAKE_TRY_COMPILE_TARGET_TYPE "STATIC_LIBRARY")
+        set(CMAKE_CXX_FLAGS "/EHsc")
```

Comment:
Please don't do this, most likely there is something wrong with your compilation environment. If you need custom flags, just pass them to cmake while configuring...

Reply:
> Please don't do this, most likely there is something wrong with your compilation environment. If you need custom flags, just pass them to cmake while configuring...

This is the content of the Windows in order to use the clang-cl to compile options, and there are some differences between MSVC, only in this way can we build triton clang-cl project success.
Passing it to cmake works, But if in cmakelist normal compilation,Will not affect other.

Reply:
I wrote an elaborate comment here to continue the discussion: https://github.com/JonathanSalwan/Triton/pull/1283#issuecomment-1708045988